provisioning/transport/http/src/TPM/TpmDelegatingHandler.cs

```diff
@@ -37,12 +37,20 @@ public TpmDelegatingHandler(SecurityProviderTpm securityProvider)
 
             if(response.StatusCode == HttpStatusCode.Unauthorized)
             {
+#if NET5_0
+                if (request.Options.TryGetValue(new HttpRequestOptionsKey<object>(ProvisioningHeaderName), out object result))
+#else
                 if (request.Properties.TryGetValue(ProvisioningHeaderName, out object result))
+#endif
                 {
                     if (result is Action<string> setSasToken)
                     {
                         string target = GetTarget(request.RequestUri.LocalPath);
+#if NET5_0
```

Comment:
Yeah seeing these further down creating a helper that handles the httpResponse ReadAsString with the CTS would go a long way.